src/resource_pool_texture.hpp

```diff
@@ -52,7 +52,7 @@ namespace wr
 		TextureHandle GetDefaultMetalic();
 		TextureHandle GetDefaultAO();
 
-		virtual Texture* GetTexture(uint64_t texture_id) = 0;
+		virtual Texture* GetTextureResource(TextureHandle handle) = 0;
```

Comment:
you could template this so it casts internally. instead of having to cast it every time you call this function.

Reply:
Also this function technically shouldn't have to be pure virtual. it could be implemented here if written well.